src/libs/Network.js

```diff
@@ -144,18 +150,25 @@ function processNetworkRequestQueue() {
         // Check to see if the queue has paused again. It's possible that a call to enhanceParameters()
         // has paused the queue and if this is the case we must return. We don't retry these requests
         // since if a request is made without an authToken we sign out the user.
-        if (!shouldMakeRequest(queuedRequest)) {
+        if (!canMakeRequest(queuedRequest)) {
             return;
         }
 
         HttpUtils.xhr(queuedRequest.command, finalParameters, queuedRequest.type)
-            .then(queuedRequest.resolve)
+            .then((response) => {
+                if (response.jsonCode === 407) {
+                    onAuthTokenExpired(queuedRequest, response);
```

Comment:
I'm kind of leaning toward having this be a little more generic. The network lib isn't really supposed to have any business logic in it and I think that knowledge about _what_ a 407 error means is bringing some of the business logic into this lib. I'm thinking of our API in Web-Expensify where you can do things like `.handle([407, 500, 418])`. 

My suggestion would be to either do:
1. rename to `on407JsonCode` (we can refactor it later if we want to handle more codes)
2. rename to `onJsonCode(407)` (feels like over-engineering at this point to handle any number of JSON codes)

Reply:
Right, that does make sense. I wonder if it is too specific to even mention a `jsonCode` if we really want to avoid the business logic. What if we changed this to `onResponse()` + `onError()` callbacks and have the `Network` lib do nothing else with `response` by default and just pass it to the callback.

Then this would end up looking like

```js
HttpUtils.xhr(queuedRequest.command, finalParameters, queuedRequest.type)
    .then(response => onResponse(queuedRequest, response))
    .catch(error => onError(queuedRequest, error))
```

Reply:
I like that! Though, what would determine an error? Would that be ONLY HTTP errors (eg. 500, 404, etc.)?

Reply:
Yep exactly the error would be basically this

https://github.com/Expensify/Expensify.cash/blob/653750d956fd035992797ed71f4cbce6f6a71edd/src/libs/HttpUtils.js#L23-L25

Reply:
And this logic can probably be moved from there and into `API`

https://github.com/Expensify/Expensify.cash/blob/653750d956fd035992797ed71f4cbce6f6a71edd/src/libs/HttpUtils.js#L26-L33

Reply:
OK, great!